source/common/http/http1/codec_impl.cc

```diff
@@ -512,7 +512,7 @@ Status ConnectionImpl::completeLastHeader() {
     RETURN_IF_ERROR(sendProtocolError(Http1ResponseCodeDetails::get().TooManyHeaders));
     const absl::string_view header_type =
         processing_trailers_ ? Http1HeaderTypes::get().Trailers : Http1HeaderTypes::get().Headers;
-    return codecProtocolError(absl::StrCat(header_type, " size exceeds limit"));
+    return codecProtocolError(absl::StrCat(header_type, " count exceeds limit"));
```

Comment:
does this suffice for your use case? if you want the difference to show up in the response (as opposed to details in logging), `error_code_` will be the response body. right now it is  `Http::Code::RequestHeaderFieldsTooLarge` for both.

Reply:
That's an excellent remark. My use case is not on requests, though, it's for responses. Where we had a response from a server sending a whooping 116 headers, the error in Envoy showed up as (1) headers disconnect before response and (2) 503 Upstream Disconnect. 

Only by increasing logging we could figure out that there were too many headers in the response, causing Envoy to terminate the connection.

Reply:
Ah! Thanks for the clarification.